dockertest/subtest.py

```diff
@@ -429,13 +458,62 @@ def try_all_stages(self, name, subsubtest):
                               self.exception_info["error_source"],
                               detail)
             exc_info = self.exception_info["exc_info"]
+            # Ignore known problems in specific docker builds
+            if self.is_known_failure(name):
+                self.final_subsubtests.add(name)
             # Treat N/A as passed, even though this may hide obscure failures
             if isinstance(detail, TestNAError):
                 self.logwarning("Treating TestNAError as PASS")
                 self.final_subsubtests.add(name)
             # cleanup() will still be called before this propagates
             raise exc_info[0], exc_info[1], exc_info[2]
 
+    def is_known_failure(self, subsubtestname):
+        """
+        Do we have a registered known failure in this subtest when running
+        on the currently-installed docker? Return True if so.
+        Side effect: log warning messages to help human debuggers.
+        """
+        # e.g. docker_cli/subtest/subsubtest
+        fullname = os.path.join(self.config_section, subsubtestname)
+        known = known_failures()
```

Comment:
Perhaps it's premature optimization, but e.g. the "help" test has MANY subsubtests.  As the CSV file grows larger (and/or system is under memory pressure) performance will take a big hit waiting for the disk inside a loops of loops.  One idea could be to cache the known_failures, something like:

```python
class SubSubtest(subtestbase.SubBase):
    ...
    #: For internal-use only, don't touch it!
    _kf_cache = None
    ...
    def is_known_failure(self, subsubtestname):
        ...
        if self._kt_cache is None:
            self.kt_cache = known_failures()
        known = self.kt_cache
        ...
```